index.js

```diff
@@ -101,7 +101,7 @@ function asPromise(opts) {
 
 					res.body = data;
 
-					if (opts.json && statusCode !== 204) {
+					if (opts.json && res.body) {
```

Comment:
Does it make sense to return the unparsed body?
If I'm not mistaken if the status code is `204` and there is a body we only prevent it from being parsed.